### DISCUSSION TITLE
A feature to automatically expand the Server[i] -> Databases[j] -> Schemas[k] -> Tables tree.

### DISCUSSION THREAD

**Discussion by hyperion-cs** [dbeaver/dbeaver#20686] [Q&A]
**Is your feature request related to a problem? Please describe.**
I guess I'm not the only one who is sick of manually expanding the tree via `Server[i] -> Databases -> Databases[j] -> Schemas -> Schemas[k] -> Tables` every time, when it's often possible to do it automatically. That's as much as 6 clicks. 6 clicks, Carl!

**Describe the solution you'd like**
A fairly common case is when you have only one database in the `Databases` path on the server, and only one schema (`public`) inside `Schemas`. On top of that, you very often want to go to `Tables` in that schema. In this case, why not provide a way to quickly expand the tree along this path, given that you need access there extremely often (as a rule)? In case you have several databases, it is possible to make automatic expansion to `Tables` in the schema `public` after selecting a particular database.
This functionality can be implemented in the context menu (click on PCM), for example. Or by holding some key like Ctrl/etc. It is also possible by adding a new type of bookmark - "typical paths in the Server tree" (about this below).

**Describe alternatives you've considered**
As far as I know, now it is only possible to expand the tree to `Tables` manually.
In addition, we can add `Bookmarks` to specific paths of **specific** servers (after that we can use the "Navigate To" function), but that's not really what I'm talking about. This solves part of the problem, but you have to add a bookmark for each server. This is not very convenient. However, the requested feature could be implemented in a way similar to `Bookmarks`, but adding "typical paths" to it, which can then be applied to **any** server.

**Additional context**
I am talking about the `Database Navigator` panel.

**Comment by E1izabeth** [comment]
Thank you for the idea.

**Comment by LonwoLonwo** [comment]
Hello @hyperion-cs 

You can use these settings for automatic tree expanding:

![2023-07-25 10_06_20-Preferences](https://github.com/dbeaver/dbeaver/assets/45152336/72cba774-44d2-427c-8e58-91f859cade9b)

**Comment by hyperion-cs** [comment]
@LonwoLonwo, thank you! That really helped. You may close this issue if no one else objects.
